### ISSUE TITLE
login promise should be rejected upon failed login

### ISSUE THREAD

**Issue by movitto** [elpheria/rpc-websockets#79]
Greetings! Thanks for the great library, starting to put it to use in a new project.

While experimenting w/ the authentication feature, I noticed the client 'login' promise will always result the 'resolve' callback being invoked even if setAuth returns false. The client can distinguish between successful and failed logins via the parameter specified to this callback (true for success, false for failure) but I feel it would be more inline w/ asynchronous javascript if the 'reject' callback was invoked on failure.

eg in the current situation the client needs to be implemented like so:

```javascript
ws.login({username, password}).then(function(result) {
  if(result)
    ; // handle successful login
  else
    ; // handle failed login
})
```

If implemented via resolve/reject, the client could handle the two cases like so:

```javascript
ws.login({username, password}).then(function(result) {
  ; // handle success login

}).catch(function(err){
  ; // handle failed login
})
```

Thoughts?

**Comment by mkozjak** [comment]
Hey, @movitto! Thanks for the kind words!

Yes, I completely agree on this one. I believe we should implement a try/catch block at https://github.com/elpheria/rpc-websockets/blob/f99c046b01b2389df96f6ad1899996343c2f3742/src/lib/client.ts#L179

Care to provide a PR? Thank you in advance!

**Comment by mkozjak** [comment]
This would also require a major version bump since it breaks client’s api.

**Comment by mkozjak** [comment]
@movitto can you checkout branch v7.x and check if the implementation is good enough for you?

**Comment by mkozjak** [comment]
It works.

```js
var WebSocket = require('./dist/index.js').Client
var WebSocketServer = require('./dist/index.js').Server

// instantiate Server and start listening for requests
var server = new WebSocketServer({
  port: 8080,
  host: 'localhost'
})

// register an RPC method
server.register('sum', function(params) {
  return params[0] + params[1]
})

// ...and maybe a protected one also
server.register('account', function() {
  return ['confi1', 'confi2']
}).protected()

server.setAuth(function() {
    if (arguments['0'].password !== "foobar")
        return false

    return true
})

// instantiate Client and connect to an RPC server
var ws = new WebSocket('ws://localhost:8080')

ws.on('open', function() {
    ws.login({'username': 'confi1', 'password':'foobar'})
    .then(function() {
        console.log("login successful")

        ws.call("account").then(function(r) {
            console.log(r)
        })
    })
    .catch(function(error) {
        console.log("login error:", error.message)
    })
})
```

**Comment by mkozjak** [comment]
The feature will be available in v7.0.0.

**Comment by movitto** [comment]
Thanks @mkozjak, haven't gotten the chance to try it out locally yet but it looks great!
